cs-vmware/src/main/java/io/cloudslang/content/vmware/entities/CustomExecutor.java

```diff
@@ -0,0 +1,42 @@
+package io.cloudslang.content.vmware.entities;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Created by giloan on 10/5/2016.
+ */
+public class CustomExecutor {
+
+    private final ThreadPoolExecutor executor;
+    private final boolean parallel;
+
+    public CustomExecutor(boolean parallel) {
+        this.parallel = parallel;
+        if (parallel) {
+            this.executor = (ThreadPoolExecutor) Executors.newCachedThreadPool();
+        } else {
+            this.executor = null;
+        }
+    }
+
+    public void execute(final Runnable task) {
+        if (parallel) {
+            executor.execute(task);
+        } else {
+            task.run();
+        }
+    }
+
+    public void shutdown() throws InterruptedException {
+        if (parallel) {
+            executor.shutdown();
+            executor.awaitTermination(Long.MAX_VALUE, TimeUnit.SECONDS);
```

Comment:
The shutdown method is called after all threads have started to upload vmdk disk files. The timeout value is so large because the files themselves are very large and network connectivity could be slow leading to a slow update, but an update that will eventually complete the transfer.